smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -208,6 +208,17 @@ contract MixinPurchase is
       revert INSUFFICIENT_VALUE();
     }
 
+    // if params have changed, then update them
+    if(_originalPrices[_tokenId] != inMemoryKeyPrice) {
```

Comment:
Do we even need the `if` statement? (would it save some bytecode to not have them)